azure blob storage module - closes #205 (#784)
bob-crutchley/courseware: topics/azure/modules/blob-storage/README.md

# Blob Storage

<!--TOC_START-->
## Contents
- [Overview](#overview)
- [Unstructured Data](#unstructured-data)
- [Blob storage design](#blob-storage-design)
- [Blob Storage Access](#blob-storage-access)
- [Tutorial](#tutorial)
	- [New Resource Group](#new-resource-group)
	- [Storage Account](#storage-account)
	- [Adding a Container](#adding-a-container)
	- [Clean Up](#clean-up)
- [Exercises](#exercises)

<!--TOC_END-->
## Overview

Blob storage is Microsoft's **object storage** solution for the cloud.

It is optimised for storing **massive** amounts of **unstructured** data.

## Unstructured Data

Unstructured data is data that doesn't conform to a particular data model or definition.

Examples of this are text data and binary data.

> ## Blob storage design

Reviewer: ```suggestion
## Blob Storage Design
```

## Tutorial

In this tutorial we will be adding some `blob` storage to our Azure account.

### New Resource Group

The first thing is to create a new resource group, which we can do by clicking the burger bar in the top left corner and choosing `Resource groups`:

![RG](https://i.imgur.com/sQW6O8U.png)

Then, we can click `+Add`:

![RG2](https://i.imgur.com/iyE2TTe.png)

Here, we can name the resource group and pick a location to have it in:

![RG3](https://i.imgur.com/A02i9WX.png)

Lastly, we will click `Review & Create` and then `Create` to create the resource group:

![RG4](https://i.imgur.com/zMF2A9R.png)

### Storage Account

Next we will create a `storage account`.

Type `Storage Accounts` in the search bar and click it when it appears.

Now click **+Add**.

In the field enter the following:

- `Resource Group` - The resource group we created
- `Storage Account Name` - Anything you like, must be **Unique**, can only contain lowercase letters and numbers
- `Location` - UK South

Leave the rest as default. And click **Review and Create**.

Then when the validations pass, click **Create**.

### Adding a Container

> In the left pane select **Containers** under `blob services`.

Reviewer: ```suggestion
In the left pane select **Containers** under `Blob service`.
```

Now click on your container and click **Upload**.

Now a window will appear with the option to select a file.
Select any image file (You may have to download them).
Then click **Upload**.

![file-select](https://lh3.googleusercontent.com/htg-G2ihZRJhvPUJnLP_EzvgmvVyBHvnLgfKZMNbocOqSFu77SIms9X9UJmUvgsIhRcTtWFOcO1DmPKWGb1RGOOcofP0bjwPj1V5TIKQIKpNjrl9cmGFrsB1vq_T8YLPYxDhzfKW0ER68yUPkXnj5znUPyM28ji86zF7mxxt4uDlJ-gN2b8E2PHvq3Azs9UUYQ6T7fUMS36KP4xv65_K4K3y-DDY3MKgig9Txc478dprVG0VeOlaEnWCdSXDtifNUoqOaqt2hBDu6-79f-qXVrxZOhwZsDZHY_7DfFnFLQwM_-fmqxhKPq72I_R2iErqUhIZAxFfpc6Yap4gkAc2vC8fpQnufrY76Ldx59HJszqS0bOSGT_Xd9kKsK1OILuA9IfYpN60mjwydEowniH1tTPncxZvvKpA2udyZvxOpxGPCD_1TALx87nz9TYpWeFYZbH1slPVroI8_QoWR1vVb_1bd6H8imjBV-VOZJ4IZEh7hrR5KtxWcxnA9pE4Z8UMHSjNUUX-ztthBQOAbNfiZTS1Ul9ggkVX-sAT7e1RixzdLZ2x1WDPLcowiL72nO0M8Nw5DiCmRcWN2LmW17HWtEQcogLE5lHnSK5tU-3BIn4spKU0HAbTDn3iZIvNd0jtRHPDVrGf1327q54gbDGekXJQszPKY9IiO3z6_wgzyCBexSj6QypS4pWui8hWj2Wynfru8hNOPDMyI8tm1EP09yuH5CAHdCQOy8ET_wmRGmBOYa4=w403-h283-no)

You have now added some blob storage to your storage account.

> 

Reviewer: Maybe add some info about how to find the link so they can see the image on the web.

> 

Author: ```suggestion

Now if you click on the image you have uploaded and select **Edit**, you will see the image. 
```

- `Storage Account Name` - Anything you like, must be **Unique**, can only contain lowercase letters and numbers
- `Location` - UK South

Leave the rest as default. And click **Review and Create**.

Then when the validations pass, click **Create**.

### Adding a Container

In the left pane select **Containers** under `Blob service`.

![containers](https://lh3.googleusercontent.com/YG11I6rMCWkfxA6ImDMSi9YYyGV_dLnU3Feetcjozs6i04-_cfpy0utvuJN43jFnZuekFSxuFqgRiDcFXHiUMfgfLm_oZ_bJINFIb7TYpxIO_O_atkoEn-dth-GrF-vBUMvu586GnSNtJGK7zPgjKHoemkf7jJpSIKB3F9OIVMbKsxEzwh0jD8AX0P7DMjxSQ9H_EagZNRvfXkP0WjYzSUuxIqcCJGmweGmdU1sPNXHJSZ4JVf6g63B3e_0Iw3G_MGLpZlCxEej7NLEsSl6JS7nRnaDkwLaq-jNZkDuZSg8tInjhA1gngUXNE38E1UPelVtt_UTHKSOhotwEunKfUulrFK3TcztsZwQFcmMi4ZatYHqTnn1aQW_4NKnMbquI6oTEqiVPg8mbZm1vtTRPxIL8C5jIgfFQAueuJn4oi7WcOxVOYtm7MAf2zZZlHZIHNbgXczsYfaIWoQobuaNYv_3zBjgo-aGlQKJrnw6Jvmt6gge9JGfafiq-7pt9UAk8Yl0UPQt4UNxRsgn9bP8xGdLpm-krEQ9XlcOTn6VKHIY7SfpBVqcwRGIPqjPwz60erda2PBI-r9tNWIt8pZqgvDhIPHvg43JVIK8q-6ck9n5WQVIBBDq17j6a0YYdaahjJcvYadeX2iKVMwB_Y8c3ndBMB_lQThCI7AQt46PT3gY2SQZbDxeclwgvBCLcokvsZJyEqsh_0c8q_ZeqQeqeZAjM1mNiAnx7DVf7VWhzZrraUPw=w240-h248-no)

> Then enter the name **blob-container**. And click **Okay**.

Reviewer: ```suggestion
Then enter the name **blob-container**. And click **Ok**.
```

> ![container-name](https://lh3.googleusercontent.com/4962gbXU-KzS--A-HDQ92FpRz8JerStBvYbyOxBEJCOsITuwGTJ05hFKSoxWbNHLrZpA7byDnKK3rZu2HBZ8zmTx6RP5dAedRUIDCJr5UQg0sqooMK9AFM4ImxVDWS1uj7BWEg58Hqf7LpSszoNMNe2A2OkyowSewv3S_oy8uIwUeAo3gmau0MAg_EkcKcbWgCuFAgs0yKzB6kZQhAajCaATNXH0w_GB-Ehzk8F-DflUfr6o8HozyQCAxivvlsrhiVkL9PZlIzQLAaW_eNCQAkSXWtWF6bjGPhBaTyk7Bd9S1jn6dUNsnuRgAwzTj6tv_vbrvdBdfmFmzBBAPnAK_MoVzsGJof_zRvVegJnsvjCdfTYUXJw_wIOUcR1SJyYP5abFFN-uwrnEL_whoJtIZbXb03d9xTER5Ds7hK6UCMPM6oZwaDP2EhkfLm20rGY9U1GEcLb6_sTPrpyk8cl3lTXrw_ThnfhybBopPRw0peW5hmbkVFOTBxPSXmsQVGanpUsKdfZflh1usfbxyUsC94dgaF2XBkd_nvwoRLqVYxqqkCMXlq1DhQck2K6NDnWOKePPXrdD66OuTHMmDNdeckXBZcawD67KtvDkFvGTA14Ap7A89pRKmPz4mMFaf9SJNpX4FSj5ZR9eZAGC2bMTeK8W03fEn7MYatCXxfM7gbpc_ZJ3iBmbFOEWvaY9LxbLNqWm136nKyIKLLCZWyRho11JuhiXymCoyOV73angWAZNLAM=w684-h252-no)

Reviewer: [![Image from Gyazo](https://i.gyazo.com/6f349fe996ae9e0fdd0a6bb2b24f711d.png)](https://gyazo.com/6f349fe996ae9e0fdd0a6bb2b24f711d)

Now click on your container and click **Upload**.

Now a window will appear with the option to select a file.
Select any image file (You may have to download them).
Then click **Upload**.

![file-select](https://lh3.googleusercontent.com/htg-G2ihZRJhvPUJnLP_EzvgmvVyBHvnLgfKZMNbocOqSFu77SIms9X9UJmUvgsIhRcTtWFOcO1DmPKWGb1RGOOcofP0bjwPj1V5TIKQIKpNjrl9cmGFrsB1vq_T8YLPYxDhzfKW0ER68yUPkXnj5znUPyM28ji86zF7mxxt4uDlJ-gN2b8E2PHvq3Azs9UUYQ6T7fUMS36KP4xv65_K4K3y-DDY3MKgig9Txc478dprVG0VeOlaEnWCdSXDtifNUoqOaqt2hBDu6-79f-qXVrxZOhwZsDZHY_7DfFnFLQwM_-fmqxhKPq72I_R2iErqUhIZAxFfpc6Yap4gkAc2vC8fpQnufrY76Ldx59HJszqS0bOSGT_Xd9kKsK1OILuA9IfYpN60mjwydEowniH1tTPncxZvvKpA2udyZvxOpxGPCD_1TALx87nz9TYpWeFYZbH1slPVroI8_QoWR1vVb_1bd6H8imjBV-VOZJ4IZEh7hrR5KtxWcxnA9pE4Z8UMHSjNUUX-ztthBQOAbNfiZTS1Ul9ggkVX-sAT7e1RixzdLZ2x1WDPLcowiL72nO0M8Nw5DiCmRcWN2LmW17HWtEQcogLE5lHnSK5tU-3BIn4spKU0HAbTDn3iZIvNd0jtRHPDVrGf1327q54gbDGekXJQszPKY9IiO3z6_wgzyCBexSj6QypS4pWui8hWj2Wynfru8hNOPDMyI8tm1EP09yuH5CAHdCQOy8ET_wmRGmBOYa4=w403-h283-no)

You have now added some blob storage to your storage account.

> Now if you click on the image you have uploaded and select **Edit**, you will see the image. 

Reviewer: ```suggestion
Now if you click on the image you have uploaded, copy the `URL` and paste it in a new window, you will see the image you just added.
```

> 

Reviewer: ```suggestion

[![Image from Gyazo](https://i.gyazo.com/fdedbc5f3df8dccc13b32d4bbe89f747.png)](https://gyazo.com/fdedbc5f3df8dccc13b32d4bbe89f747)

```
